app/scripts/controllers/mmi-controller.ts

```diff
@@ -848,28 +849,30 @@ export default class MMIController extends EventEmitter {
   async setAccountAndNetwork(origin: string, address: string, chainId: number) {
     await this.appStateController.getUnlockPromise(true);
     const addressToLowerCase = address.toLowerCase();
-    const { address: selectedAddress } = this.messenger.call(
+    const { address: selectedAddress } = this.messagingSystem.call(
       'AccountsController:getSelectedAccount',
     );
 
     if (selectedAddress.toLowerCase() !== addressToLowerCase) {
-      const internalAccount = this.messenger.call(
+      const internalAccount = this.messagingSystem.call(
         'AccountsController:getAccountByAddress',
         addressToLowerCase,
       );
-      this.messenger.call(
-        'AccountsController:setSelectedAccount',
-        internalAccount.id,
-      );
+      if (internalAccount) {
+        this.messagingSystem.call(
+          'AccountsController:setSelectedAccount',
+          internalAccount.id,
+        );
+      }
     }
 
     const selectedChainId = getCurrentChainId({
-      metamask: this.networkController.state,
+      metamask: this.#networkControllerState,
     });
 
     if (selectedChainId !== toHex(chainId)) {
       const networkConfiguration =
-        this.networkController.state.networkConfigurationsByChainId[
+        this.#networkControllerState.networkConfigurationsByChainId[
           toHex(chainId)
         ];
```

Comment:
We can also use a messenger action for this:

``` suggestion
      const networkConfiguration = this.messagingSystem.call(
        'NetworkController:getNetworkConfigurationByChainId',
        toHex(chainId),
      );
```

Interestingly, I found that the type for this action is not exported from `@metamask/network-controller`. We can fill this type in for now if we add this above `AllowedActions` in `shared/constants/mmi-controller.ts`. This should allow us to add it to the type union.

``` typescript
type NetworkControllerGetNetworkConfigurationByChainId = {
  type: `NetworkController:getNetworkConfigurationByChainId`;
  handler: NetworkController['getNetworkConfigurationByChainId'];
};
```